File: cpp/p28.cpp
#include "euler.h"

using namespace std;

int main() {
  int sum = 1;

  for( int n = 3; n <= 1001; n += 2 ) {
    vector<long long> corners = get_n_by_n_square_corners( n );
	
    sum += accumulate( corners.begin(), corners.end(), 0 );
  }
	
  cout << sum << endl;
}

File: clojure/p28.clj
(ns p28
  (:use [euler :only (sqr sum timed-test)]))

(defn corners [n]
  (for [i (range 4)]
    (- (sqr n) (* i (dec n)))))

(timed-test
 669171001
 (let [sizes (range 3 (inc 1001) 2)
       all-corners (mapcat corners sizes)]
   (apply + 1 all-corners)))
